File: Year 1/Semester 2/Java/Lessons/classes/discussion.java
package classes;

public class discussion {
    public static void main(String[] args) {

        // object 1
        university micheal = new university("tuks", 1900);
        micheal.speak();

        // object 2
        university matthew = new university("su", 1899);
        matthew.speak();

        // object 3
        university boble = new university("uct", 1910);
        boble.speak();

        // call it again !duplication
        micheal.speak();

        // 
        int x = micheal.getYear();
        System.out.println(x);

        // change the age
        micheal.setYear(2400);
        micheal.speak();

        // boble.graduate();  cant call this as its a private method!
    }
}

File: Year 1/Semester 1/C/Lessons/classes/discussion.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char name[50];
    int year;
} University;

// Function to print university details
void speak(University u) {
    printf("University: %s, Established Year: %d\n", u.name, u.year);
}

// Getter for the year
int getYear(University u) {
    return u.year;
}

// Setter for the year
void setYear(University *u, int newYear) {
    u->year = newYear;
}

// Private method (can't be called from main, only shown here for example)
// static void graduate(University u) {
//     printf("%s has graduated!\n", u.name);
// }

int main() {
    // object 1
    University micheal = {"tuks", 1900};
    speak(micheal);

    // object 2
    University matthew = {"su", 1899};
    speak(matthew);

    // object 3
    University boble = {"uct", 1910};
    speak(boble);

    // call it again !duplication
    speak(micheal);

    // Get year of establishment
    int x = getYear(micheal);
    printf("%d\n", x);

    // Change the year of establishment
    setYear(&micheal, 2400);
    speak(micheal);

    // boble.graduate();  Can't call this, it's private in Java and commented in C
    return 0;
}
